### DISCUSSION TITLE
Ignore a file or directory

### DISCUSSION THREAD

**Discussion by interfect** [detekt/detekt#4825] [Q&A] [answered]
I'm working on a fork of a project that uses Detekt in its build system. I don't want to rip out Detekt, but I do want to exclude certain files or directories from being processed by Detekt (because these files are stub files that I need to add to the project to make it build without some libraries it expects, and they aren't real code, and therefore don't need e.g. doc comments). It looks like in `detekt.yml` I can set an `excludes` for each rule; is there a way to set a global `excludes`? Or to put a `@file:Suppress` that will suppress all Detekt processing rather than just certain rules?

**Comment by cortinico** [comment, marked as answer]
> It looks like in `detekt.yml` I can set an `excludes` for each rule; is there a way to set a global `excludes`? Or to put a `@file:Suppress` that will suppress all Detekt processing rather than just certain rules?

You can put an `excludes` at the ruleset level inside the config file. So you won't have to copy it over to every rule.

If you're using the CLI you can use the `--excludes` parameter to exclude certain paths.
If you're using the Gradle Plugin, similarly you can configure the `source` and `excludes` for the Detekt tasks

**Reply by Lysander086** [reply]
To validate my understanding,  I can configure ```excludes [ '**/wanted/**' ]```under comments, complexity, coroutines and such. Correct?

**Reply by BraisGabin** [reply]
Yes

**Reply by govindrai** [reply]
> You can put an excludes at the ruleset level inside the config file. So you won't have to copy it over to every rule.

This is missing from the [documentation](https://detekt.dev/docs/introduction/configurations)? 

What is the ruleset level? Is there a top level exclude you can avoid having to add the excludes under comments, complexity, coroutines and such

**Reply by BraisGabin** [reply]
Yes

```yaml
complexity:
  excludes: [...] # This will be applied to all the rules inside complexity
```

Any PR improving the documentation is more than welcome :)

**Reply by govindrai** [reply]
That didn't work for me. I got a schema error.  The [json schema](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/detekt-1.22.0.json#L248) doesn't actually support a top-level `excludes` key for each ruleset. Instead you can only specify the `excludes` property for each rule under a ruleset.

**Reply by cortinico** [reply]
If that's the case, then it's a bug and we should fix it.

**Reply by BraisGabin** [reply]
#4162 seems the fix for this, right?

**Reply by cortinico** [reply]
Yeah essentially. Still I'm unsure how is the JSON schema generated (I supposed is done from converting a yml file and inferring from it). So something to potentially look into 🤔

**Reply by BraisGabin** [reply]
I think that we need a good refactor on `detekt-generator`. We should be able to generate that schema from that.
